trio/_tests/test_exports.py

```diff
@@ -106,28 +109,34 @@ def no_underscores(symbols):
         if sys.implementation.name != "cpython":
             pytest.skip("mypy not installed in tests on pypy")
 
-        # mypy behaves strangely when passed a huge semicolon-separated line with `-c`
-        # so we use a tmpfile
-        tmpfile = tmpdir / "check_mypy.py"
-        tmpfile.write_text(
-            f"import {modname}\n"
-            + "".join(f"{modname}.{name}\n" for name in runtime_names),
-            encoding="utf8",
-        )
+        cache = Path.cwd() / ".mypy_cache"
         from mypy.api import run
 
-        mypy_res = run(["--config-file=", "--follow-imports=silent", str(tmpfile)])
-
-        # check that there were no errors (exit code 0), otherwise print the errors
-        assert mypy_res[2] == 0, mypy_res[0]
+        # This pollutes the `empty` dir. Should this be changed?
+        run(["--config-file=", "--cache-dir=./.mypy_cache", "-c", f"import {modname}"])
```

Comment:
Do you really need to run mypy for each submodule? I tested it quickly and it seems to populate cache for the whole package when just running `import trio`.

So could e.g. do something like having a global variable for having generated the cache this test run, and use that both here and in `test_static_tool_sees_class_members`.

Reply:
```bash
$ pwd
~/Git/trio
$ touch trio/py.typed
$ pip install .
$ mkdir empty; cd empty
$ mypy --config-file= --cache-dir=.mypy_cache -c "import trio"
$ l .mypy_cache/3.11/trio | wc -l
61
```

Reply:
A global variable that tracks that the cache is updated once per pytest invocation is plenty, and on my machine validating the cache takes 0.5s - so saving the extra mypy calls means a total of ~14 seconds of time saved, which makes quite the difference imo.